olwebsite-app/src/contexts/UserContext.jsx

```diff
@@ -0,0 +1,45 @@
+import  React, { useContext, useEffect, useState } from 'react';
```

Comment:
Checkout the following link for our jsx file format standards. Helps to keep things consistent across all the components. 

[https://github.com/oneleif/olWebsite-React/issues/47](url)

Reply:
Good to know. Not sure if you're saying that there is something I need to correct though? I looked through it and didn't see anything glaring that stood out to me

Reply:
Not exactly a correction, but can you add the sectional comment boxes in there? 
Also, I was thinking we could refactor and export the login, logout, and register functions as well, making this more of an all around authentication context. What do you think?

Reply:
Hmm i see what you're saying but I'm not sure if those should be grouped with this. The context itself is just using hooks to access what is in essence a global variable to all of its children components. So I'd think the only things we'd want in this context file would be getters for values associated with that variable (in this case the user object)

Reply:
Went ahead and added the sectional comment boxes 👍

Reply:
I understand what you mean. I was thinking references to those functions but not the actual implementations. The thinking behind that is that the context would be the only part of the application actually interacting with that global variable. We sort of have that already with the setUser function exported by useUser. So instead of having a random register function deal with directly calling setUser we would expose a reference to register() from the userProvider. All interactions with the user/token/session objects would have to go through the userProvider.

Reply:
Great work!

Reply:
Ahh ok then yea that would make sense if you have all functionality that interacts with the setUser to be within the UserContext file, and then don't include setUser in the useUser hook. 💯